op-proposer/proposer/driver.go

```diff
@@ -309,33 +389,56 @@ func (l *L2OutputSubmitter) loop() {
 
 	ctx := l.ctx
 
-	ticker := time.NewTicker(l.Cfg.PollInterval)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-ticker.C:
-			output, shouldPropose, err := l.FetchNextOutputInfo(ctx)
-			if err != nil {
-				break
+	if l.dgfContract == nil {
+		ticker := time.NewTicker(l.Cfg.PollInterval)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				output, shouldPropose, err := l.FetchNextOutputInfo(ctx)
+				if err != nil || !shouldPropose {
+					break
+				}
+
+				l.proposeOutput(ctx, output)
+			case <-l.done:
+				return
 			}
-			if !shouldPropose {
-				break
-			}
-			cCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
-			if err := l.sendTransaction(cCtx, output); err != nil {
-				l.Log.Error("Failed to send proposal transaction",
-					"err", err,
-					"l1blocknum", output.Status.CurrentL1.Number,
-					"l1blockhash", output.Status.CurrentL1.Hash,
-					"l1head", output.Status.HeadL1.Number)
-				cancel()
-				break
+		}
+	} else {
```

Comment:
Wdyt about extracting this code into two separate loop functions like `loopL2OO()` and `loopDGF()` and calling them here?

And is the new DGF loop behavior tested?

Reply:
Broke them out 👍 

The new DGF loop behavior has been tested locally, but not covered in `op-e2e` just yet. Since the bridge doesn't currently rely on the `DisputeGameFactory`, we don't yet have a purpose for this in the devnet. This is mostly preparatory work :)